### PR TITLE
Use some Compat utilities

### PR DESCRIPTION
**Pull request by TacoTheDank**
They do checks internally.

### CODE REVIEW

File: app/src/main/java/me/magnum/melonds/ui/shortcutsetup/ShortcutSetupActivity.kt [hunk 1]
```diff
@@ -58,14 +46,13 @@ class ShortcutSetupActivity : AppCompatActivity() {
             putExtra(EmulatorActivity.KEY_URI, rom.uri.toString())
         }
         val romIcon = viewModel.getRomIcon(rom)
-        val shortcutInfo = ShortcutInfo.Builder(this, rom.uri.toString())
+        val shortcutInfo = ShortcutInfoCompat.Builder(this, rom.uri.toString())
                 .setShortLabel(rom.name)
-                .setIcon(Icon.createWithBitmap(buildShortcutBitmap(romIcon)))
+                .setIcon(IconCompat.createWithBitmap(buildShortcutBitmap(romIcon)))
```
Comment (rafaelvcaetano) [inline app/src/main/java/me/magnum/melonds/ui/shortcutsetup/ShortcutSetupActivity.kt:51]: Not sure why this is happening only now, but when clicking on a ROM to create a shortcut, nothing happens and the following error is printed to the Logcat:

`E/JavaBinder: !!! FAILED BINDER TRANSACTION !!!  (parcel size = 1049868)`

Looks like there is a size limit of 1MB in Binder transactions and the size that the icon is taking is... exactly 1MB (512 x 512 x 4). You can read more about this limit here: https://developer.android.com/reference/android/os/TransactionTooLargeException.html

So, one of the fixes here is to reduce the size of the icon's bitmap. Now that I think about it, 512x512 is way to large to be displayed in such a small area. As such, I believe that the size of the icon can be reduced to 256x256 without compromises.

File: app/src/main/java/me/magnum/melonds/MelonDSApplication.kt [hunk 2]
```diff
@@ -35,13 +32,12 @@ class MelonDSApplication : Application(), Configuration.Provider {
     }
 
     private fun createNotificationChannels() {
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
-            return
-        }
+        val defaultChannel = NotificationChannelCompat.Builder(
+            NOTIFICATION_CHANNEL_ID_BACKGROUND_TASKS, NotificationManagerCompat.IMPORTANCE_LOW)
```
Comment (rafaelvcaetano) [inline app/src/main/java/me/magnum/melonds/MelonDSApplication.kt:36]: Everything seems fine here. I'll just ask you to put the parameters in the same line as the constructor call.

```kotlin
val defaultChannel = NotificationChannelCompat.Builder(NOTIFICATION_CHANNEL_ID_BACKGROUND_TASKS, NotificationManagerCompat.IMPORTANCE_LOW)
            .setName(getString(R.string.notification_channel_background_tasks))
            ...
```

**Review by rafaelvcaetano — CHANGES_REQUESTED** [review]
Everything looks mostly fine. There's just a problem when selecting a ROM for the shortcut widget that needs to be addressed before accepting this.

And a minor formatting nit-pick.

File: app/src/main/java/me/magnum/melonds/ui/shortcutsetup/ShortcutSetupActivity.kt (hunk 1 above)
Comment (TacoTheDank) [reply app/src/main/java/me/magnum/melonds/ui/shortcutsetup/ShortcutSetupActivity.kt:51]: Sounds good to me! Will change

File: app/src/main/java/me/magnum/melonds/MelonDSApplication.kt (hunk 2 above)
Comment (TacoTheDank) [reply app/src/main/java/me/magnum/melonds/MelonDSApplication.kt:36]: Sure thing!

**Comment by TacoTheDank** [comment]
@rafaelvcaetano Done!

**Review by rafaelvcaetano — APPROVED** [review]

**Pull request merged**